### PR TITLE
feat(capture): remove team resolution code

### PR DESCRIPTION
**Pull request by xvello**
## Problem

Lightweight capture has been running for a month now, so the team resolution path is now dead code. Let's remove it to make the real code path more explicit.

## Changes

- Remove all occurences of `team_id` from `capture.py` and the dead code branches, including: 
  - `log_event_to_dead_letter_queue`, as it was only used in case PG was down. Kafka being down generates 503s (service unavailable).
  - `EventIngestionContext` and its associated methods and tests from utils
  - `_ensure_web_feature_flags_in_properties` (deprecation was announced twice on Slack)
- Rename the misleadingly named functions
- Update `person.py` to submit events with a token instead of a team_id. This change ensures that these events will be co-located with the distinct_id's other events
- Update tests, and remove the ones testing removed features:
  - `test_capture_event_ip_anonymize` handled in plugin-server-ingestion now
  - `test_personal_api_key`, `test_personal_api_key_from_batch_request` and `test_batch_request_with_invalid_auth`: this auth method has been deprecated (used by old versions of the zapier integration)
  - `test_batch_incorrect_token` -> `test_batch_incorrect_token_shape`
  - `test_add_feature_flags_if_missing` feature flag value backfill has been deprecated
  - `test_database_unavailable`: we don't hit PG anymore

## How did you test this code?

- Unit and e2e test coverage

- Manual tests:
  - local instance with autocapture works fine (both events and recordings)
  - adding and removing properties from the app (person.py) work fine

### CODE REVIEW

File: posthog/api/utils.py [hunk 1]
```diff
@@ -212,140 +208,6 @@ def get_data(request):
...
-    `posthog.models.Team`, which includes many variables that are not specific
-    to the context of ingestion. With this structure we can be deliberate about
-    our ingestion interfaces.
-
-    The initial driver for this was to reduce the amount of data we were
-    fetching from postgresql db.
-    """
-
-    team_id: int
-    anonymize_ips: bool
```
Comment (fuziontech) [inline posthog/api/utils.py:232]: 🔥

**Review by fuziontech — APPROVED** [review]
Looks good to me!
🚢 it

File: posthog/api/capture.py [hunk 2]
```diff
@@ -58,7 +55,7 @@
 EVENTS_DROPPED_OVER_QUOTA_COUNTER = Counter(
     "capture_events_dropped_over_quota",
     "Events dropped by capture due to quota-limiting, per resource_type, team_id and token.",
```
Comment (tomasfarias) [inline posthog/api/capture.py:57]: nit: This `Counter` no longer uses `team_id`.

**Review by tomasfarias — APPROVED** [review]
Looks good, nice to see all those lines dropping :heavy_minus_sign:

**Pull request merged**